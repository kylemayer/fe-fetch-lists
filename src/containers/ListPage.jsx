import React, { Component } from 'react';
import style from './List.css';
import VillagerList from '../components/displays/VillagerList';
export default class ListPage extends Component {
  render() {
    return (
      <div className={style.list}>
        <h3>Villagers from the village</h3>
        <VillagerList
          villagers={[
            {
              id: 1,
              name: 'Apple',
              image: 'test',
              species: 'test',
              birthday: 'old',
              skill: 'fast',
              phrase: 'atta boy'
            }
          ]}
        />
	   </div>
    );
  }
}
