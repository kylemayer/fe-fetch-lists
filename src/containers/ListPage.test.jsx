import React from 'react';
import { render, screen, cleanup } from '@testing-library/react';
import ListPage from './ListPage';
import { MemoryRouter } from 'react-router-dom';

describe('List Page', () => {
  afterEach(() => cleanup());
  it('should render the List page', async () => {
    render(
      <MemoryRouter>
        <ListPage />
      </MemoryRouter>
    );

    const ul = await screen.findByRole('list', { name: 'villagers' });
    expect(ul).toBeInTheDocument();
    expect(ul).toMatchSnapshot();
  });
});
