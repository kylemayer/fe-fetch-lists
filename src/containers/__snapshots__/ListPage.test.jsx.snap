// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List Page should render the List page 1`] = `
<ul
  aria-label="villagers"
>
  <li>
    <a
      href="/5f5fb4bbbfd05c2aed82e460"
    >
      <h3>
        Admiral
      </h3>
      <img
        alt="Admiral"
        class="Villager"
        src="https://vignette.wikia.nocookie.net/animalcrossing/images/e/ed/Admiral_NH.png/revision/latest?cb=20200802081138"
      />
      <p>
        bird
      </p>
      <p>
        January 27th (Aquarius)
      </p>
      <p>
        Writing about pickles
      </p>
      <p>
        aye aye
      </p>
    </a>
  </li>
  <li>
    <a
      href="/5f5fb4bbbfd05c2aed82e461"
    >
      <h3>
        Agent S
      </h3>
      <img
        alt="Agent S"
        class="Villager"
        src="https://vignette.wikia.nocookie.net/animalcrossing/images/a/a7/Agent_S_NH.png/revision/latest?cb=20200719092526"
      />
      <p>
        squirrel
      </p>
      <p>
        July 2nd (Cancer)
      </p>
      <p>
        Limboing
      </p>
      <p>
        sidekick
      </p>
    </a>
  </li>
  <li>
    <a
      href="/5f5fb4bbbfd05c2aed82e462"
    >
      <h3>
        Agnes
      </h3>
      <img
        alt="Agnes"
        class="Villager"
        src="https://vignette.wikia.nocookie.net/animalcrossing/images/4/42/Agnes_NH.png/revision/latest?cb=20200804230432"
      />
      <p>
        pig
      </p>
      <p>
        April 21st (Taurus)
      </p>
      <p>
        Cramming for tests
      </p>
      <p>
        snuffle
      </p>
    </a>
  </li>
  <li>
    <a
      href="/5f5fb4bbbfd05c2aed82e463"
    >
      <h3>
        Al
      </h3>
      <img
        alt="Al"
        class="Villager"
        src="https://vignette.wikia.nocookie.net/animalcrossing/images/c/c4/Al_NH.png/revision/latest?cb=20200802150354"
      />
      <p>
        gorilla
      </p>
      <p>
        October 18th (Libra)
      </p>
      <p>
        Jumping rope
      </p>
      <p>
        hoo hoo ha CFayyyeee NH
      </p>
    </a>
  </li>
  <li>
    <a
      href="/5f5fb4bbbfd05c2aed82e464"
    >
      <h3>
        Alfonso
      </h3>
      <img
        alt="Alfonso"
        class="Villager"
        src="https://vignette.wikia.nocookie.net/animalcrossing/images/e/e4/Alfonso_NH.png/revision/latest?cb=20200817164717"
      />
      <p>
        alligator
      </p>
      <p>
        June 9th (Gemini)
      </p>
      <p>
        Fishing
      </p>
      <p>
        it'sa me
      </p>
    </a>
  </li>
  <li>
    <a
      href="/5f5fb4bbbfd05c2aed82e465"
    >
      <h3>
        Alice
      </h3>
      <img
        alt="Alice"
        class="Villager"
        src="https://vignette.wikia.nocookie.net/animalcrossing/images/e/e1/Alice_NH.png/revision/latest?cb=20200803181400"
      />
      <p>
        koala
      </p>
      <p>
        August 19th (Leo)
      </p>
      <p>
        Photography
      </p>
      <p>
        guvnor
      </p>
    </a>
  </li>
  <li>
    <a
      href="/5f5fb4bbbfd05c2aed82e46c"
    >
      <h3>
        Ankha
      </h3>
      <img
        alt="Ankha"
        class="Villager"
        src="https://vignette.wikia.nocookie.net/animalcrossing/images/5/56/Ankha_NH.png/revision/latest?cb=20200817172044"
      />
      <p>
        cat
      </p>
      <p>
        September 22nd (Virgo)
      </p>
      <p>
        Writing about Pickles
      </p>
      <p>
        me meow
      </p>
    </a>
  </li>
  <li>
    <a
      href="/5f5fb4bbbfd05c2aed82e466"
    >
      <h3>
        Alli
      </h3>
      <img
        alt="Alli"
        class="Villager"
        src="https://vignette.wikia.nocookie.net/animalcrossing/images/8/8e/Alli_NH.png/revision/latest?cb=20200802061731"
      />
      <p>
        alligator
      </p>
      <p>
        November 8th (Scorpio)
      </p>
      <p>
        Yo-Yoing
      </p>
      <p>
        graaagh
      </p>
    </a>
  </li>
  <li>
    <a
      href="/5f5fb4bbbfd05c2aed82e46b"
    >
      <h3>
        Anicotti
      </h3>
      <img
        alt="Anicotti"
        class="Villager"
        src="https://vignette.wikia.nocookie.net/animalcrossing/images/8/80/Anicotti_NH.png/revision/latest?cb=20200802164003"
      />
      <p>
        mouse
      </p>
      <p>
        February 24th (Pisces)
      </p>
      <p>
        Oversleeping
      </p>
      <p>
        cannoli
      </p>
    </a>
  </li>
  <li>
    <a
      href="/5f5fb4bbbfd05c2aed82e46a"
    >
      <h3>
        Angus
      </h3>
      <img
        alt="Angus"
        class="Villager"
        src="https://vignette.wikia.nocookie.net/animalcrossing/images/5/5f/Angus_NH.png/revision/latest?cb=20200802083758"
      />
      <p>
        bull
      </p>
      <p>
        April 30th (Taurus)
      </p>
      <p>
        Doing handstands
      </p>
      <p>
        macmoo
      </p>
    </a>
  </li>
  <li>
    <a
      href="/5f5fb4bbbfd05c2aed82e467"
    >
      <h3>
        Amelia
      </h3>
      <img
        alt="Amelia"
        class="Villager"
        src="https://vignette.wikia.nocookie.net/animalcrossing/images/a/ab/Amelia_NH.png/revision/latest?cb=20200817171545"
      />
      <p>
        eagle
      </p>
      <p>
        November 19th (Scorpio)
      </p>
      <p>
        Playing poker
      </p>
      <p>
        "eaglet" (GCN)"cuz"(WW & CF & NL)
      </p>
    </a>
  </li>
  <li>
    <a
      href="/5f5fb4bbbfd05c2aed82e46d"
    >
      <h3>
        Annalisa
      </h3>
      <img
        alt="Annalisa"
        class="Villager"
        src="https://vignette.wikia.nocookie.net/animalcrossing/images/f/f0/Annalisa_NH.png/revision/latest?cb=20200802063308"
      />
      <p>
        anteater
      </p>
      <p>
        February 6th (Aquarius)
      </p>
      <p>
        Waking up early
      </p>
      <p>
        gumdrop
      </p>
    </a>
  </li>
  <li>
    <a
      href="/5f5fb4bbbfd05c2aed82e46f"
    >
      <h3>
        Antonio
      </h3>
      <img
        alt="Antonio"
        class="Villager"
        src="https://vignette.wikia.nocookie.net/animalcrossing/images/d/d4/Antonio_NH.png/revision/latest?cb=20200802063352"
      />
      <p>
        anteater
      </p>
      <p>
        October 20th (Libra)
      </p>
      <p>
        Playing horseshoes
      </p>
      <p>
        honk
      </p>
    </a>
  </li>
  <li>
    <a
      href="/5f5fb4bbbfd05c2aed82e46e"
    >
      <h3>
        Annalise
      </h3>
      <img
        alt="Annalise"
        class="Villager"
        src="https://vignette.wikia.nocookie.net/animalcrossing/images/c/c4/Annalise_NH.png/revision/latest?cb=20200802153943"
      />
      <p>
        horse
      </p>
      <p>
        December 2nd (Sagittarius)
      </p>
      <p>
        Brainteasers
      </p>
      <p>
        nipper (NL, NH)whatever (AF+, AC, AFe+)
      </p>
    </a>
  </li>
  <li>
    <a
      href="/5f5fb4bbbfd05c2aed82e468"
    >
      <h3>
        Anabelle
      </h3>
      <img
        alt="Anabelle"
        class="Villager"
        src="https://vignette.wikia.nocookie.net/animalcrossing/images/d/d8/Anabelle_NH.png/revision/latest?cb=20200802063200"
      />
      <p>
        anteater
      </p>
      <p>
        February 16th (Aquarius)
      </p>
      <p>
        Jumping rope
      </p>
      <p>
        snorty
      </p>
    </a>
  </li>
  <li>
    <a
      href="/5f5fb4bbbfd05c2aed82e470"
    >
      <h3>
        Apollo
      </h3>
      <img
        alt="Apollo"
        class="Villager"
        src="https://vignette.wikia.nocookie.net/animalcrossing/images/2/28/Apollo_NH.png/revision/latest?cb=20200719092927"
      />
      <p>
        eagle
      </p>
      <p>
        July 4th (Cancer)
      </p>
      <p>
        Staying up late
      </p>
      <p>
        pah
      </p>
    </a>
  </li>
  <li>
    <a
      href="/5f5fb4bbbfd05c2aed82e469"
    >
      <h3>
        Anchovy
      </h3>
      <img
        alt="Anchovy"
        class="Villager"
        src="https://vignette.wikia.nocookie.net/animalcrossing/images/e/ea/Anchovy_NH.png/revision/latest?cb=20200802081231"
      />
      <p>
        bird
      </p>
      <p>
        March 4th (Pisces)
      </p>
      <p>
        Remembering stuff
      </p>
      <p>
        chuurp
      </p>
    </a>
  </li>
  <li>
    <a
      href="/5f5fb4bbbfd05c2aed82e471"
    >
      <h3>
        Apple
      </h3>
      <img
        alt="Apple"
        class="Villager"
        src="https://vignette.wikia.nocookie.net/animalcrossing/images/5/57/Apple_NH.png/revision/latest?cb=20200802151508"
      />
      <p>
        hamster
      </p>
      <p>
        September 24th (Libra)
      </p>
      <p>
        Karaoke
      </p>
      <p>
        cheekers
      </p>
    </a>
  </li>
  <li>
    <a
      href="/5f5fb4bbbfd05c2aed82e472"
    >
      <h3>
        Astrid
      </h3>
      <img
        alt="Astrid"
        class="Villager"
        src="https://vignette.wikia.nocookie.net/animalcrossing/images/d/d5/Astrid_NH.png/revision/latest?cb=20200802154954"
      />
      <p>
        kangaroo
      </p>
      <p>
        September 8th (Virgo)
      </p>
      <p>
        Karaoke
      </p>
      <p>
        my pet
      </p>
    </a>
  </li>
  <li>
    <a
      href="/5f5fb4bbbfd05c2aed82e473"
    >
      <h3>
        Audie
      </h3>
      <img
        alt="Audie"
        class="Villager"
        src="https://vignette.wikia.nocookie.net/animalcrossing/images/1/1b/Audie_NH.png/revision/latest?cb=20200829012735"
      />
      <p>
        wolf
      </p>
      <p>
        August 31st (Virgo)
      </p>
      <p>
        Unknown
      </p>
      <p>
        foxtrot
      </p>
    </a>
  </li>
  <li>
    <a
      href="/5f5fb4bbbfd05c2aed82e474"
    >
      <h3>
        Aurora
      </h3>
      <img
        alt="Aurora"
        class="Villager"
        src="https://vignette.wikia.nocookie.net/animalcrossing/images/5/59/Aurora_NH.png/revision/latest?cb=20200202215217"
      />
      <p>
        penguin
      </p>
      <p>
        January 27th (Aquarius)
      </p>
      <p>
        Making Budgets
      </p>
      <p>
        b-b-baby
      </p>
    </a>
  </li>
  <li>
    <a
      href="/5f5fb4bbbfd05c2aed82e475"
    >
      <h3>
        Ava
      </h3>
      <img
        alt="Ava"
        class="Villager"
        src="https://vignette.wikia.nocookie.net/animalcrossing/images/c/cc/Ava_NH.png/revision/latest?cb=20200609055741"
      />
      <p>
        chicken
      </p>
      <p>
        April 28th (Taurus)
      </p>
      <p>
        Peeling apples
      </p>
      <p>
        beaker
      </p>
    </a>
  </li>
  <li>
    <a
      href="/5f5fb4bbbfd05c2aed82e476"
    >
      <h3>
        Avery
      </h3>
      <img
        alt="Avery"
        class="Villager"
        src="https://vignette.wikia.nocookie.net/animalcrossing/images/d/dd/Avery_NH.png/revision/latest?cb=20200608222354"
      />
      <p>
        eagle
      </p>
      <p>
        February 22nd (Pisces)
      </p>
      <p>
        Skimming books
      </p>
      <p>
        skree-haw
      </p>
    </a>
  </li>
  <li>
    <a
      href="/5f5fb4bbbfd05c2aed82e478"
    >
      <h3>
        Baabara
      </h3>
      <img
        alt="Baabara"
        class="Villager"
        src="https://vignette.wikia.nocookie.net/animalcrossing/images/d/df/Baabara_NH.png/revision/latest?cb=20200802165552"
      />
      <p>
        sheep
      </p>
      <p>
        March 28th (Aries)
      </p>
      <p>
        Drawing circles
      </p>
      <p>
        daahling
      </p>
    </a>
  </li>
  <li>
    <a
      href="/5f5fb4bbbfd05c2aed82e47b"
    >
      <h3>
        Barold
      </h3>
      <img
        alt="Barold"
        class="Villager"
        src="https://vignette.wikia.nocookie.net/animalcrossing/images/9/9d/Barold_NH.png/revision/latest?cb=20200802141140"
      />
      <p>
        cub
      </p>
      <p>
        March 2nd (Pisces)
      </p>
      <p>
        Computing
      </p>
      <p>
        cubby
      </p>
    </a>
  </li>
</ul>
`;
